public/app/plugins/datasource/elasticsearch/elastic_response.ts

```diff
@@ -516,7 +535,40 @@ export class ElasticResponse {
       }
     }
 
-    return { data: seriesList };
+    return { data: seriesList, key: this.targets[0]?.refId };
+  };
+
+  processPPLResponseToSeries = () => {
+    const target = this.targets[0];
+    const response = this.response;
+    // We check if any valid date type is contained in the response
+    const timeFieldIndex = _.findIndex(
+      response.schema,
+      (field: { type: string }) =>
+        field.type === 'timestamp' || field.type === 'datetime' || field.type === 'date' || field.type === 'time'
+    );
+    const valueIndex = timeFieldIndex === 0 ? 1 : 0;
```

Comment:
why timeFieldIndex === 0? the field of type timestamp could in any position of response.schema, right?